vpc/service/generate_assignment_id.go

```diff
@@ -148,6 +148,7 @@ func (vpcService *vpcService) generateAssignmentID(ctx context.Context, req getE
 
 	ass := &assignment{
 		trunk:            req.trunkENI,
+		branch:           &data.BranchENI{}, // Will be populated in populateAssignment
```

Comment:
Is it so bad that it is nil at first?

Why not add this as part of the populateAssignmentUsingAlreadyAttachedENI() function? 

```
populateAssignmentUsingAlreadyAttachedENI(ass) {
 
ass.branch = &data.BranchENI{}
pg.bla....
```

Reply:
I added it here just in case there are things in between this initialization and `populateAssignmentUsingAlreadyAttachedENI` that use `ass.branch`. Though there should be none for now.

Added another check in `populateAssignmentUsingAlreadyAttachedENI` before the usage, too.